prog/dftb+/lib_dftbplus/parser.F90

```diff
@@ -3711,6 +3716,96 @@ subroutine readSolvGB(node, geo, input)
   end subroutine readSolvGB
 
 
+  subroutine readSolvSASA(node, geo, input)
+
+    !> Node to process
+    type(fnode), pointer :: node
+
+    !> Geometry of the current system
+    type(TGeometry), intent(in) :: geo
+
+    !> Contains the input for the dispersion module on exit
+    type(TSASAInput), intent(out) :: input
+
+    type(string) :: buffer, modifier
+    type(fnode), pointer :: child, value1, value2, field, child2, dummy
+    integer :: iSp, ii, gridPoints
+    real(dp) :: conv
+    real(dp), allocatable :: vdwRadDefault(:)
+
+    if (geo%tPeriodic) then
+      call detailedError(node, "SASA model currently not available with PBCs")
+    end if
+
+    call getChildValue(node, "ProbeRadius", input%probeRad, modifier=modifier, child=field)
+    call convertByMul(char(modifier), lengthUnits, field, input%probeRad)
+
+    call getChildValue(node, "Smoothing", input%smoothingPar, 0.3_dp*AA__Bohr, &
+        & modifier=modifier, child=field)
+    call convertByMul(char(modifier), lengthUnits, field, input%smoothingPar)
+
+    call getChildValue(node, "Tolerance", input%tolerance, 1.0e-6_dp, child=child)
+
+    call getChildValue(node, "AngularGrid", gridPoints, 230, child=child)
+    input%gridSize = 0
+    do ii = 1, size(gridSize)
+      if (gridPoints == gridSize(ii)) then
+        input%gridSize = ii
+      end if
+    end do
+    if (input%gridSize == 0) then
+      call detailedError(child, "Illegal number of grid points for numerical integration")
```

Comment:
Usability change for a later pull request would be to suggest the grid nearest to the request value.

Reply:
Using a bisection search would automatically include this.

Reply:
OK, so might as well then. If you've not found the included implementation, see lib_math/bisect.F90

Reply:
Now this can only happen for 5 or less requested grid points.